kafka-connect/src/main/java/com/hazelcast/jet/contrib/connect/KafkaConnectSources.java

```diff
@@ -0,0 +1,193 @@
+     * offsets, it will restore the partition offsets and resume the consumption
+     * from where it left off.
+     *
+     * @param properties Kafka connect properties
+     * @return a source to use in {@link com.hazelcast.jet.pipeline.Pipeline#readFrom(StreamSource)}
+     */
+    public static StreamSource<SourceRecord> connect(Properties properties) {
+        String name = properties.getProperty("name");
+        return SourceBuilder.timestampedStream(name, ctx -> new Context(ctx, properties))
+                            .fillBufferFn(Context::fillBuffer)
+                            .createSnapshotFn(Context::createSnapshot)
+                            .restoreSnapshotFn(Context::restoreSnapshot)
+                            .destroyFn(Context::destroy)
+                            .build();
+    }
+
+    private static class Context {
+
+        private final SourceConnector connector;
+        private final SourceTask task;
+        private final Map<String, String> taskConfig;
+
+        private Map<Map<String, ?>, Map<String, ?>> partitionsToOffset = new HashMap<>();
+        private boolean taskInit;
+
+        Context(Processor.Context ctx, Properties properties) {
+            try {
+                // inject hazelcast.instance.name for retrieving from JVM instance factory in the Debezium source
+                if (properties.containsKey("database.history")) {
+                    JetInstance jetInstance = ctx.jetInstance();
+                    String instanceName = HazelcastInstanceFactory.getInstanceName(jetInstance.getName(),
+                            jetInstance.getHazelcastInstance().getConfig());
+                    properties.setProperty("database.history.hazelcast.instance.name", instanceName);
+                }
+                String connectorClazz = properties.getProperty("connector.class");
+                Class<?> connectorClass = Thread.currentThread().getContextClassLoader().loadClass(connectorClazz);
+                connector = (SourceConnector) connectorClass.getConstructor().newInstance();
+                connector.initialize(new JetConnectorContext());
+                connector.start((Map) properties);
+
+                taskConfig = connector.taskConfigs(1).get(0);
+                task = (SourceTask) connector.taskClass().getConstructor().newInstance();
+
+            } catch (Exception e) {
+                throw rethrow(e);
+            }
+        }
+
+        void fillBuffer(TimestampedSourceBuffer<SourceRecord> buf) {
+            if (!taskInit) {
+                task.initialize(new JetSourceTaskContext());
+                task.start(taskConfig);
+                taskInit = true;
+            }
+            try {
+                List<SourceRecord> records = task.poll();
+                if (records == null) {
+                    return;
+                }
+
+                for (SourceRecord record : records) {
+                    long ts = record.timestamp() == null ? 0 :
```

Comment:
can be a 1 liner `long ts = record.timestamp() == null ? 0 : record.timestamp();`